plasma_framework/test/src/framework/ExitGameController.test.js

```diff
@@ -0,0 +1,268 @@
+            await expectRevert(
+                this.controller.processExits(fakeNonAddedTokenAddress, 0, 1),
+                "Such token has not be added to the plasma framework yet"
+            );
+        });
+
+        it('rejects when the exit queue is empty', async () => {
+            await expectRevert(
+                this.controller.processExits(this.dummyToken, 0, 1),
+                "Exit queue is empty"
+            );
+        });
+
+        it('rejects when the "top unique priority" mismatch with the specified one', async () => {
+            await this.dummyExitGame.enqueue(0, this.dummyToken, this.dummyExit);
+
+            const nonExistingUniqeuePriority = 123;
+            await expectRevert(
+                this.controller.processExits(this.dummyToken, nonExistingUniqeuePriority, 1),
+                "Top unique priority of the queue is not the same as the specified one"
+            );
+        });
+
+        it('processes when the "top unique priority" is set to 0', async () => {
+            await this.dummyExitGame.enqueue(0, this.dummyToken, this.dummyExit);
+
+            const tx = await this.controller.processExits(this.dummyToken, 0, 1);
+            await expectEvent.inLogs(tx.logs, 'ProcessedExitsNum', {
+                processedNum: new BN(1),
+                token: this.dummyToken
+            });
+        });
+
+        it('processes when the "top unique priority" is set to the exact top one of the queue', async () => {
+            await this.dummyExitGame.enqueue(0, this.dummyToken, this.dummyExit);
+            const uniquePriority = await this.dummyExitGame.uniqueuPriorityFromEnqueue();
+
+            const tx = await this.controller.processExits(this.dummyToken, uniquePriority, 1);
+            await expectEvent.inLogs(tx.logs, 'ProcessedExitsNum', {
+                processedNum: new BN(1),
+                token: this.dummyToken
+            });
+        });
+
+        it('calls the "processExit" function of the exit processor when processes', async () => {
+            await this.dummyExitGame.enqueue(0, this.dummyToken, this.dummyExit);
+
+            await this.controller.processExits(this.dummyToken, 0, 1);
+
+            // 'exitIdProcessed' would be set when processExit is called in the DummyExitGame
+            const exitIdProcessed = await this.dummyExitGame.exitIdProcessed();
+            expect(exitIdProcessed).to.be.bignumber.equal(new BN(this.dummyExit.exitId));
+        });
+
+        it('deletes the processed exit from queue after processed', async () => {
+            await this.dummyExitGame.enqueue(0, this.dummyToken, this.dummyExit);
+
+            await this.controller.processExits(this.dummyToken, 0, 1);
+
+            const priorityQueueAddress = await this.controller.exitsQueues(this.dummyToken);
+            const priorityQueue = await PriorityQueue.at(priorityQueueAddress);
+            expect(await priorityQueue.currentSize()).to.be.bignumber.equal(new BN(0));
```

Comment:
Like in previous comment. I think it would be better to test that through controller API. 
Check whether calling `processExits` for the second time fails with expected error message.

Reply:
The test description here is:
>deletes the processed exit from queue after processed

It feels to me more straight forward by checking priority queue directly here. Check priority queue empty via re-call the function and would fail relies on the knowledge of "the function would fail when the priority queue is empty". Feel like one more step of implementation knowledge to reach there actually. 

How do you think?

Reply:
You are right on that. Test matches the description. But I have a feeling that we could test on a higher level. The things we care about is that exits are processed, none is processed twice, they are processed with respect to their priority, limit numbers and arguments values have effect. Priority queue underneath is an implementation detail. But on the other hand maybe that matters in security critical components, like smart contracts. Let's see what @paulperegud thinks of that.

Reply:
By checking current size we are checking if size has changed, not if value is no longer in the queue. I would just try to `getMin()` next value out of the queue and observe call getting reverted by EVM.

Also - please check the old tests for priority queue. They cover a lot.

Reply:
@pgebal I see your point now. I like your description about the things we care about:
> exits are processed, none is processed twice, they are processed with respect to their priority

I changed the test to be testing
> processes with the order of priority and delete the processed exit from queue

So I first put two things to the queue and check after process the top (`getMin`) is the lower priority one + the size decreases.